tests/delete_regress/models.py

```diff
@@ -151,3 +148,38 @@ class OrderedPerson(models.Model):
 
     class Meta:
         ordering = ["name"]
+
+
+# Models for #35073
+
+
+class Owner(models.Model):
+    name = models.CharField(max_length=32)
+
+    def __str__(self):
+        return self.name
+
+
+def get_default_owner():
+    return Owner.objects.get_or_create(name="default")[0]
+
+
+class Pet(models.Model):
+    owner = models.ForeignKey(Owner, models.SET(get_default_owner))
+
+
+class Owner2(models.Model):
+    name = models.CharField(max_length=32)
+
+    def __str__(self):
+        return self.name
+
+
+def get_default_owner2():
+    return Owner2.objects.get_or_create(name="default2")[0]
+
+
+class Pet2(models.Model):
+    owner = models.ForeignKey(
+        Owner2, default=get_default_owner2, on_delete=models.SET_DEFAULT
+    )
```

Comment:
`Owner` and `Owner2` are unnecessary, please reuse existing models.

Reply:
We also don't need to `Pet` models. I'd use:
```diff
diff --git a/tests/delete_regress/models.py b/tests/delete_regress/models.py
index cbe6fef334..f7a19d38e4 100644
--- a/tests/delete_regress/models.py
+++ b/tests/delete_regress/models.py
@@ -72,6 +72,25 @@ class Eaten(models.Model):
     meal = models.CharField(max_length=20)
 
 
+def get_best_toy():
+    toy, _ = Toy.objects.get_or_create(name="best")
+    return toy
+
+
+def get_worst_toy():
+    toy, _ = Toy.objects.get_or_create(name="worst")
+    return toy
+
+
+class Collector(models.Model):
+    best_toy = models.ForeignKey(
+        Toy, default=get_best_toy, on_delete=models.SET_DEFAULT, related_name="toys"
+    )
+    worst_toy = models.ForeignKey(
+        Toy, models.SET(get_worst_toy), related_name="bad_toys"
+    )
+
+
 # Models for #15776
 
 

```

Reply:
Updated